hazelcast/src/main/java/com/hazelcast/sql/impl/operation/QueryOperation.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.operation;
+
+import com.hazelcast.internal.util.UUIDSerializationUtil;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Base class for query operations.
+ */
+public abstract class QueryOperation implements IdentifiedDataSerializable {
+
+    public static final int PARTITION_ANY = -1;
+
+    private UUID callerId;
+
+    protected QueryOperation() {
+        // No-op.
+    }
+
+    public UUID getCallerId() {
+        return callerId;
+    }
+
+    public void setCallerId(UUID callerId) {
+        this.callerId = callerId;
+    }
+
+    public int getPartition() {
+        return PARTITION_ANY;
+    }
+
+    protected int getPartitionForHash(int hash) {
```

Comment:
Doesn't look like it's returning a partition, more like a partition hash code. Used like a utility function, should be static probably. Some javadoc may help in any case.

Reply:
This function converts an arbitrary integer to a positive integer, which is then interpreted as a logical partition. 

The notion of "partition" is used here to provide an analogy with our partition pool, which also uses logical partitions [1], not physical. 

[1] com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl#getPartitionThreadId(int, int)

Reply:
Additional JavaDocs are added.